### PR TITLE
fix Using a reference for the variable on range scope

### PR DESCRIPTION
**Pull request by s3rj1k**
Signed-off-by: s3rj1k <evasive.gyron@gmail.com>

### CODE REVIEW

**Comment by s3rj1k** [comment]
https://github.com/sourcegraph/jsonrpc2/issues/29

File: jsonrpc2.go [hunk 1]
```diff
@@ -644,8 +644,13 @@ func (m *anyMessage) UnmarshalJSON(data []byte) error {
 		if len(msgs) == 0 {
 			return errors.New("jsonrpc2: invalid empty batch")
 		}
-		for _, msg := range msgs {
-			if err := checkType(&msg); err != nil {
+		for i := range msgs {
+			if err := checkType(&msg{
```
Comment (unknwon) [inline jsonrpc2.go]: In a second thought... why don't we just use, do scopelint complain about this:
```suggestion
			if err := checkType(msgs[i]); err != nil {
```

Comment (s3rj1k) [reply jsonrpc2.go]: @unknwon checkType whants pointer to ```msg```

Comment (unknwon) [reply jsonrpc2.go]: What about `&msgs[i]`, does scopelint complain this?

Comment (s3rj1k) [reply jsonrpc2.go]: @unknwon yep, last one works, ```&msgs[i]```

Comment (s3rj1k) [reply jsonrpc2.go]: but I am actually not sure that this is a good idea, pointer to a value inside slice that can move its values inside memory, can this actually be an issue in this case?

Comment (unknwon) [reply jsonrpc2.go]: If `checkType` is read-only, I think that's fine. But yeah, what's in the PR feels safer.

**Comment by unknwon** [comment]
I'm merging it, since it's easy to revert if something goes wrong.

**Comment by s3rj1k** [comment]
@unknwon :+1:

**Pull request merged**